File: app/src/main/java/com/example/kotlinnotes/views/TodoView.java
package com.example.kotlinnotes.views;

import android.content.Context;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.widget.CheckBox;
import android.widget.TextView;
import androidx.constraintlayout.widget.ConstraintLayout;
import com.example.kotlinnotes.R;
import com.example.kotlinnotes.models.Todo;

public class TodoView extends ConstraintLayout {

    private CheckBox cbComplete;
    private TextView tvDescription;


    public TodoView(Context context) {
        super(context);
    }

    public TodoView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public TodoView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void initView(Todo todo) {
        cbComplete = findViewById(R.id.cb_complete);
        tvDescription = findViewById(R.id.tv_description);
        tvDescription.setText(todo.getDescription());
        cbComplete.setChecked(todo.isComplete());
        if (todo.isComplete()) {
            createStrikeThrough();
        }
        setUpCheckStateListener();
    }

    public void setUpCheckStateListener() {
        cbComplete.setOnCheckedChangeListener((button, isChecked) ->{
            if (isChecked) {
                createStrikeThrough();
            } else {
                removeStrikeThrough();
            }
        });
    }

    private void createStrikeThrough() {
        tvDescription.setPaintFlags(tvDescription.getPaintFlags()| Paint.STRIKE_THRU_TEXT_FLAG);
    }

    private void removeStrikeThrough() {
        tvDescription.setPaintFlags(tvDescription.getPaintFlags() & ~(Paint.STRIKE_THRU_TEXT_FLAG));
    }


}

File: app/src/main/java/com/example/kotlinnotes/views/TodoView.kt
package com.example.kotlinnotes.views

import android.content.Context
import android.graphics.Paint
import android.util.AttributeSet
import androidx.constraintlayout.widget.ConstraintLayout
import com.example.kotlinnotes.models.Todo
import kotlinx.android.synthetic.main.view_todo.view.*

class TodoView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 1
) : ConstraintLayout(context, attrs, defStyleAttr) {

    fun initView(todo: Todo, callback: ((Boolean) -> Unit)?) {
        tv_description.text = todo.description
        cb_complete.isChecked = todo.isComplete
        if (todo.isComplete) {
            tv_description.setStrikeThrough()
        }
        setUpCheckStateListener(todo, callback)
    }

    fun setUpCheckStateListener(todo: Todo, callback: ((Boolean) -> Unit)? = null) {
        cb_complete.setOnCheckedChangeListener { _, isChecked ->
            todo.isComplete = isChecked
            callback?.invoke(isChecked)
            if (isChecked) {
                tv_description.setStrikeThrough()
            } else {
                tv_description.removeStrikeThrough()
            }
        }
    }
}
